src/main/java/com/surrealdb/refactor/driver/parsing/JsonElementParser.java

```diff
@@ -0,0 +1,61 @@
+package com.surrealdb.refactor.driver.parsing;
+
+import java.net.ProtocolException;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.surrealdb.refactor.exception.UnhandledProtocolResponse;
+import com.surrealdb.refactor.types.QueryResult;
+
+public class JsonElementParser {
+	//this method parses the JSON element regardless of whether it is a single element
+	// or an array of elements
+	
+	public static QueryResult[] parseJsonElement(JsonElement outerResultJson) throws ProtocolException {
```

Comment:
By having this method static there are a lot of restrictions being implied
- it cannot easily be configured via the class (say, for example, by choosing the JSON parsing implementation)
- it cannot be changed to an interface to allow multiple implementations

Furthermore, the method name contains implementation details. The implementation could likely change in the future, but we would still want to keep this method. Perhaps `parseResultMessage` would be better?

Also, I would like to avoid having explicit exceptions. I don't think explicit exception tests in the code are a nice developer experience.

Reply:
sounds good. I will implement the changes. Thanks for all the feedback. I feel like I am learning a lot.

Reply:
That's awesome to hear! Really happy about that :) It's great code overall, so I really hope the feedback isn't off-putting. It's getting close to being able to land.